File: 1-moderate/column-names/main.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <sstream>

int OFFSET = (int) 'A';

std::string convertToExcelColumnName(int number)
{
    number--;
    std::vector<char> result;

    while (number >= 0) {
        int quotient = number / 26;
        int remainder = number % 26;
        result.push_back((char) (OFFSET + remainder));
        number = quotient - 1;
    }

    std::stringstream ss;

    for(std::vector<char>::reverse_iterator it = result.rbegin(); it != result.rend(); ++it) {
        ss << *it;
    }

    return ss.str();
}

int main(int argc, char** argv)
{
    std::ifstream file(argv[1]);

    std::string line;
    bool first = true;
    while (std::getline(file, line))
    {
        std::istringstream reader(line);
        int val;
        reader >> val;

        std::cout << convertToExcelColumnName(val) << std::endl;
    }
}

File: 1-moderate/column-names/main.py
import sys
import math

OFFSET = ord('A')
ALPHABET_LENGTH = 26

def convert_to_excel_column_name(number):
    number -= 1

    result = []
    while number >= 0:
        quotient, remainder = divmod(number, ALPHABET_LENGTH)
        result.append(chr(OFFSET + remainder))
        number = quotient - 1
    return ''.join(reversed(result))

def main(input_filename):
    test_cases = open(input_filename, 'r')
    for test in test_cases:
        test = test.strip()
        if len(test) == 0:
            continue
        number = int(test)
        print(convert_to_excel_column_name(number))
    test_cases.close()

if __name__ == '__main__':
    if len(sys.argv) != 2:
        print("Usage: python %s <input_filename>" % sys.argv[0])
        sys.exit(1)
    main(sys.argv[1])
